### PR TITLE
Cleanup of the USER_HOME variable in ``pylint.config``

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

Remove what I believe is dead code. There's more to do in ``pylint.config.__init__``  but I'm taking it slow as it's untested and hard to test right now. The refactor aims to make this more testable.

### CODE REVIEW

File: pylint/config/__init__.py [hunk 1]
```diff
@@ -18,27 +19,26 @@
...
     "Option",
     "OptionsManagerMixIn",
     "OptionParser",
     "OptionsProviderMixIn",
     "UnsupportedAction",
+    "PYLINTRC",
+    "USER_HOME",
 ]
 
-USER_HOME = os.path.expanduser("~")
```
Comment (Pierre-Sassoulas) [inline pylint/config/__init__.py:35]: USER_HOME equal ``os.path.expanduser("~")`` so it's never equal to ``~``.

File: pylint/config/environement_variable.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+from pylint.config.find_default_config_files import find_pylintrc
```
Comment (DanielNoord) [inline pylint/config/environement_variable.py:5]: The name of this file is misspelled 😄

**Review by DanielNoord — CHANGES_REQUESTED** [review]
Need to change the name

File: pylint/config/environement_variable.py (hunk 2 above)
Comment (Pierre-Sassoulas) [reply pylint/config/environement_variable.py:5]: Do you mean there should not be an "s" ?

Comment (DanielNoord) [reply pylint/config/environement_variable.py:5]: Isn't it `environment`?

Comment (Pierre-Sassoulas) [reply pylint/config/environement_variable.py:5]: Ha right, I thought it was about ``find_default_config_files `` !

**Review by DanielNoord — APPROVED** [review]

**Pull request merged**